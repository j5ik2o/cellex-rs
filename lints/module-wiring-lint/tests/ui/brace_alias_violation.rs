#![feature(register_tool)]
#![register_tool(module_wiring)]
#![warn(module_wiring::no_parent_reexport)]

mod child {
  pub struct A;
}

pub use child::{A as Alias};

fn main() {}
