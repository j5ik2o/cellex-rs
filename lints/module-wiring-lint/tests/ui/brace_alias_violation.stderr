warning: 再エクスポートは末端モジュールの直属親以外では禁止です
  --> $DIR/brace_alias_violation.rs:9:1
   |
LL | pub use child::{A as Alias};
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: ルール: 例外ルール: 末端モジュールの直属親のみ許可
   = note: 詳細: `as` を用いた再エクスポートは許可されていません
   = help: 末端モジュール内で元の名前を公開するか、呼び出し側で名前を付け替えてください
   = note: `#[warn(no_parent_reexport)]` on by default

warning: 1 warning emitted

